rust/common/redis/src/lib.rs

```diff
@@ -280,44 +277,103 @@ impl RedisClient {
     /// * `addr` - Redis connection string
     /// * `compression` - Compression configuration (see CompressionConfig)
     /// * `format` - Serialization format for values (see RedisValueFormat)
+    /// * `response_timeout` - Optional timeout for Redis command responses. `None` means no timeout (blocks indefinitely).
+    /// * `connection_timeout` - Optional timeout for establishing connections. `None` means no timeout (blocks indefinitely).
+    ///
+    /// # Panics
+    /// Panics if `Some(Duration::ZERO)` is passed - use `None` for no timeout instead.
```

Comment:
Panicking is solid enough, but is there any way you can forbid passing illegal values at the compiler level so that we don't have panics in production code? I know you're calling it out in the function signature, but IMO it'd be safer to just make this impossible in the first place, given that this is code that users specify, rather than a config variable.